src/Collection/CollectionTrait.php

```diff
@@ -140,6 +140,74 @@ public function some(callable $callback): bool
         return false;
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function none(callable $callback): bool
+    {
+        foreach ($this->optimizeUnwrap() as $key => $value) {
+            if ($callback($value, $key) === true) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function atMost(int $count, callable $callback): bool
```

Comment:
Instead of three methods for counting/comparisons, I wonder if there is a more generic 'count and compare' style operation we could add.

Reply:
Counting and comparing may not be as efficient as an early return inside a short-circuit loop that can exit if the condition is met without having to iterate over the entire collection. For example, to verify if at least two negative numbers exist, the count and compare approach needs to iterate over the entire collection:
```php
$isNegative = fn ($el) => $el < 0;

//  iterate over the entire collection / less intuitive
if ($collection->countBy($isNegative) >= 2) {...} 

//  iterate until condition met / more intuitive
if ($collection->atLeast(2, $isNegative) {...}
```

Reply:
just counting and comparing may not be as efficient as an early return inside a short-circuit loop that can exit if the condition is met without having to iterate over the entire collection. A more generic counting and comparison method could be like: `exists(callable $callback, int $min = 1, ?int $max = null): bool`, It fulfills your suggested requirements and can be optimized with early returns in some cases.

Reply:
Fair point about early-returns being more efficient.

We could also have a method like `compareCount(callable $callback, string $operator, int value)` this would let you do operations like `$collection->compareCount(fn ($item) => ..., '>=', 2)`. This could be implemented to support early returns.